packages/drivers/odsp-driver/src/prefetchLatestSnapshot.ts

```diff
@@ -111,9 +126,19 @@ export async function prefetchLatestSnapshot(
 				putInCache,
 				removeEntries,
 				enableRedeemFallback,
-			);
+			)
+				.then((value) => {
+					assert(!!snapshotEpoch, "prefetched snapshot should have a valid epoch");
+					snapshotContentsWithEpochP.resolve({ ...value, fluidEpoch: snapshotEpoch });
+				})
+				.catch((err) => {
+					snapshotContentsWithEpochP.reject(err);
+					throw err;
+				});
 			assert(cacheP !== undefined, 0x1e7 /* "caching was not performed!" */);
-			await cacheP;
+			// Remove it from the non persistent cache once it is cached in the persistent cache as then it will
+			// be fetched from persistent cache.
+			await cacheP.then(() => nonPersistentCache?.remove(nonPersistentCacheKey));
```

Comment:
I think this needs to be on .finally() path, not on .then() path.
I.e. we do not want to get stuck completely on first error forever. If it failes, it fails all in flight loads, but anybody can retry.